pkg/logging/logwriter.go

```diff
@@ -0,0 +1,96 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package logging
+
+import (
+	gorillaHandlers "github.com/gorilla/handlers"
+	"go.uber.org/zap"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+// LogWriter a type that can write logs to a zap.SugaredLogger object
+type LogWriter struct {
+	logger *zap.SugaredLogger
+}
+
+// GetLogWriter creates and fetches a LogWriter object
+func GetLogWriter(logger *zap.SugaredLogger) LogWriter {
+	return LogWriter{
+		logger: logger,
+	}
+}
+
+// Write writes the byte array to the logger object
+func (l LogWriter) Write(p []byte) (n int, err error) {
+	l.logger.Info(string(p))
+	return len(p), nil
+}
+
+// buildCommonLogLine builds a log entry for req in Apache Common Log Format.
+// ts is the timestamp with which the entry should be logged.
+// status and size are used to provide the response HTTP status and size.
+func buildCommonLogLine(req *http.Request, url url.URL, ts time.Time, status int, size int) []byte {
+	username := "-"
+	if url.User != nil {
+		if name := url.User.Username(); name != "" {
+			username = name
+		}
+	}
+
+	host, _, err := net.SplitHostPort(req.RemoteAddr)
+	if err != nil {
+		host = req.RemoteAddr
+	}
+
+	uri := req.RequestURI
+
+	// Requests using the CONNECT method over HTTP/2.0 must use
+	// the authority field (aka r.Host) to identify the target.
+	// Refer: https://httpwg.github.io/specs/rfc7540.html#CONNECT
+	if req.ProtoMajor == 2 && req.Method == "CONNECT" {
+		uri = req.Host
+	}
+	if uri == "" {
+		uri = url.RequestURI()
+	}
+
+	buf := make([]byte, 0, 3*(len(host)+len(username)+len(req.Method)+len(uri)+len(req.Proto)+50)/2)
+	buf = append(buf, host...)
+	buf = append(buf, " - "...)
+	buf = append(buf, username...)
+	buf = append(buf, ` "`...)
+	buf = append(buf, req.Method...)
+	buf = append(buf, " "...)
+	buf = append(buf, uri...)
```

Comment:
since the log writer only logs at the `info` level, we would end up logging the webhook `apikey` at info level. Is that expected?

Reply:
good catch. @kanchwala-yusuf wdyt? would that we severe security threat?